shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -266,21 +267,31 @@ - (void)setupChannels {
 
 - (void)maybeSetupPlatformViewChannels {
   if (_shell && self.shell.IsSetup()) {
+      
+    __block FlutterPlatformPlugin *blockmethochannel = (FlutterPlatformPlugin*)_platformPlugin.get();
     [_platformChannel.get() setMethodCallHandler:^(FlutterMethodCall* call, FlutterResult result) {
-      [_platformPlugin.get() handleMethodCall:call result:result];
+      [blockmethochannel handleMethodCall:call result:result];
     }];
 
+    __block __typeof(self)weakSelf = self;
```

Comment:
does this not need to be marked as `__weak`?

Reply:
in MRC files we should use __block for weak object